addons/gamification/models/gamification_challenge.py

```diff
@@ -645,6 +650,18 @@ def discard_challenge(self):
         sudoed.message_post(body=_("%s has refused the challenge", user.name))
         return sudoed.write({'invited_user_ids': (3, user.id)})
 
+    @api.model
+    @tools.ormcache()
+    def _get_cron_update_interval_or_default(self):
+        """Return interval in seconds between cron updates of challenges goals (data-provided cron).
+
+        Returns 86_400 if cron record was not found (= 1 day, default interval for this cron).
+        """
+        if not (cron_sudo := self.sudo().env.ref("gamification.ir_cron_check_challenge", raise_if_not_found=False)):
+            return 86_400
+
+        return timedelta(**{cron_sudo.interval_type: cron_sudo.interval_number}).total_seconds()
+
```

Comment:
Bad usage of the warlus operator, when you use a walrus inside of a conditional, it means that you intend to only use that variable inside of the conditional block. Here you are using the variable outside of that block, then the variable should be declared outside. 

Also, nitpicking, using `_` inside of numbers is welcome when it makes the number readable, `86400` is readable already but it is kind of a magic number, it would be nice to add a comment next to it (even if it's written in the docstring already)

```suggestion
        cron_sudo = self.sudo().env.ref("gamification.ir_cron_check_challenge", raise_if_not_found=False)
        if not cron_sudo:
            return 86400  # one day

        return timedelta(**{cron_sudo.interval_type: cron_sudo.interval_number}).total_seconds()

```